src/main/java/uk/gov/companieshouse/overseasentitiesapi/utils/PopulateUpdateSubmission.java

```diff
@@ -0,0 +1,163 @@
+package uk.gov.companieshouse.overseasentitiesapi.utils;
+
+import java.time.LocalDate;
+import java.util.Objects;
+import java.util.Optional;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.overseasentitiesapi.model.BeneficialOwnersStatementType;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.AddressDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.DueDiligenceDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.OverseasEntityDueDiligenceDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.OverseasEntitySubmissionDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.PresenterDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.DueDiligence;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.FilingForDate;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.Presenter;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.UpdateSubmission;
+
+/**
+ * This is a helper class for FilingsService when the filing is an update. The purpose of this class
+ * is to populate change submission values into UpdateSubmission.java with the ultimate gaol to
+ * produce a json output of change submissions. It uses OESubmissionDTO and UpdateDTO values to
+ * populate UpdateSubmission model.
+ */
+@Component
+public class PopulateUpdateSubmission {
+
+    /**
+     * Method populates values into UpdateSubmission for JSON output
+     */
+    public void populate(OverseasEntitySubmissionDto overseasEntitySubmissionDto,
+            UpdateSubmission updateSubmission) {
+
+        updateSubmission.setUserSubmission(overseasEntitySubmissionDto);
+        populateDueDiligence(overseasEntitySubmissionDto, updateSubmission);
+        populatePresenter(overseasEntitySubmissionDto, updateSubmission);
+        populateStatements(overseasEntitySubmissionDto, updateSubmission);
+        populateFilingForDate(overseasEntitySubmissionDto, updateSubmission);
+    }
+
+    private void populateDueDiligence(
+            OverseasEntitySubmissionDto overseasEntitySubmissionDto,
+            UpdateSubmission updateSubmission) {
+
+        var dueDiligenceDto = Optional.ofNullable(overseasEntitySubmissionDto.getDueDiligence());
+        var overseasEntityDueDiligenceDto = Optional.ofNullable(
```

Comment:
Can you move this initialisation within the 'else' block

Reply:
I've noticed this a few times within this file at the start (and sometimes at the end) of methods